thefuck/rules/su.py

```diff
@@ -0,0 +1,12 @@
+def match(command):
+    if command.script_parts and '&&' not in command.script_parts and command.script_parts[0] == 'su':
```

Comment:
```suggestion
    if '&&' not in command.script_parts:
```

Also, Fish Shell must be considered. Support to `&&` operator has been added to version 3. Earlier versions only support the `and` operator.

Reply:
sudo.py suffers with the same issue, and that's already released.

Reply:
That's true. I's been sitting there unnoticed. Thanks, @djh82. That must be fixed.